ext/pcntl/pcntl.c

```diff
@@ -125,6 +125,16 @@ static zend_class_entry *QosClass_ce;
 
 #define LONG_CONST(c) (zend_long) c
 
+#if defined(__linux__)
+#include <linux/wait.h>
+#endif
+#if defined (P_ALL)
+#define HAVE_P_ALL 1
```

Comment:
supporting waitid includes having P_ALL, P_PID, P_PGID guaranteed.

Reply:
I did some investigation and broke them up into:

- POSIX (incl. OpenBSD and Solaris)
- Linux (POSIX + P_PIDFD)
- NetBSD (POSIX + P_UID + P_GID + P_SID)
- FreeBSD (NetBSD + P_JAILID

Reply:
On the other hand macOS seems to be using an antiquated version of FreeBSD libc, perhaps it is a POSIX-like system that doesn't have waitid and thus this function should be optional in pcntl?

Reply:
Yes good point I forgot to mention it but that s easy enough, you need to add waitid to the list of existing check in ext/pcntl/config.m4 then guarding things (including in the stub file) with HAVE_WAITID.

Reply:
Yeah, that was pretty straightforward. Done.